lib/readline.js

```diff
@@ -22,30 +22,21 @@ const stripVTControlCharacters = internalReadline.stripVTControlCharacters;
 
 
 exports.createInterface = function(input, output, completer, terminal) {
-  var rl;
-  if (arguments.length === 1) {
-    rl = new Interface(input);
-  } else {
-    rl = new Interface(input, output, completer, terminal);
-  }
-  return rl;
+  return new Interface(input, output, completer, terminal);
 };
 
 
 function Interface(input, output, completer, terminal) {
   if (!(this instanceof Interface)) {
-    // call the constructor preserving original number of arguments
-    const self = Object.create(Interface.prototype);
-    Interface.apply(self, arguments);
-    return self;
+    return new Interface(input, output, completer, terminal);
   }
 
   this._sawReturn = false;
 
   EventEmitter.call(this);
   var historySize;
 
-  if (arguments.length === 1) {
+  if (input && input.input) {
```

Comment:
Is this change in behaviour intentional? If `input` is a custom implementation that happens to have an `input` property, then it will cause the remaining arguments to be ignored. Can we find a safer way replicating the arguments length check? Maybe the ugly/obvious `if (input && !output && !completer && !terminal)` approach would be the most appropriate?